### PR TITLE
Fixes pacman generators not working and adjusts them

### PR DESCRIPTION
**Pull request by MacBlaze1**
## About The Pull Request

This PR fixes Pacmans and makes the UI actually work along with the power usage and stuff.
## Why It's Good For The Game

This should have been fixed a long time ago, as it is not intended behavior and its a feature that was ignored. 
## Changelog
:cl:
fix: made the pacman's UI usable
fix: made the pacman actually eat away at the sheets.
balance: adjusted the pacman to be longer lasting and have more power for use
/:cl:

### CODE REVIEW

**Comment by jroinc** [comment]
You may want to look at the number of available generators on maps, and keep in mind that shock damage scales with available power in the powergrid to avoid making instagib wires.

**Comment by MacBlaze1** [comment]
> You may want to look at the number of available generators on maps, and keep in mind that shock damage scales with available power in the powergrid to avoid making instagib wires.

Is it total power or power split between sources? example, 120 kW available across 8 erazorwires, 10 kW per, 40 excess power, please tell me what is the shock power.

**Comment by MacBlaze1** [comment]
> You may want to look at the number of available generators on maps, and keep in mind that shock damage scales with available power in the powergrid to avoid making instagib wires.

Just calculated it, its 20.6 damage lmao. The minimum damage is 20, so you’re essentially adding 0.6 damage each time.

File: code/modules/power/port_gen.dm [hunk 1]
```diff
@@ -17,7 +17,7 @@
 	var/base_icon = "portgen0"
 	var/datum/looping_sound/generator/soundloop
 
-	//interaction_flags_atom = INTERACT_ATOM_ATTACK_HAND | INTERACT_ATOM_UI_INTERACT | INTERACT_ATOM_REQUIRES_ANCHORED
+	interaction_flags = INTERACT_MACHINE_TGUI
```
Comment (spookydonut) [inline code/modules/power/port_gen.dm:20]: this should only be set on /obj/machinery/power/port_gen/pacman

**Pull request merged**